apps/s_client.c

```diff
@@ -2201,6 +2204,36 @@ int s_client_main(int argc, char **argv)
             BIO_read(sbio, sbuf, BUFSIZZ);
         }
         break;
+    case PROTO_SIEVE:
+        {
+            int foundit = 0;
+            BIO *fbio = BIO_new(BIO_f_buffer());
+
+            BIO_push(fbio, sbio);
+            /* wait for multi-line response to end from Sieve */
+            do {
+                mbuf_len = BIO_gets(fbio, mbuf, BUFSIZZ);
+                /*
+                 * According to RFC 5804 § 1.7, capability
+                 * is case-insensitive, make it uppercase
+                 */
+                if (mbuf_len > 1 && mbuf[0] == '"') {
+                    make_uppercase(mbuf);
+                    if (strncmp(mbuf, "\"STARTTLS\"", 10) == 0)
+                        foundit = 1;
+                }
+            } while (mbuf_len > 1 && mbuf[0] == '"');
+            (void)BIO_flush(fbio);
+            BIO_pop(fbio);
+            BIO_free(fbio);
+            if (!foundit)
+                BIO_printf(bio_err,
+                           "Didn't find STARTTLS in server response,"
+                           " trying anyway...\n");
+            BIO_printf(sbio, "STARTTLS\r\n");
+            BIO_read(sbio, sbuf, BUFSIZZ);
```

Comment:
> I just tried to stick with what already exists for other protocols.

Well, multiple wrongs don't make it right. If anything one should overhaul rather than keeping on wrong path. Yes, you can say that it was inconsistent of me/us to approve previous request, but we're better off if it stops at some point, so why not now :-)

Basically there are two options, past reply to STARTTLS is parsed that is. a) write something on screen if reply was negative and ignore it; b) respect negative reply and exit. I would vote for a), because when we call something a protocol, we king of make am implicit commitment to follow it. One can argue that not finding STARTTLS in CAPABILITIES can be ignored, because server would still have an option to deny the explicit request, but initiating TLS handshake when server explicitly refused to do so is hardly appropriate.

I'd appreciate if somebody else from the team chimes in with opinion. If we decide in favour of the respecting negative reply in response to STARTTLS, I'd appreciate if you could file a fix for recently committed NNTP. If you decline, it's fine too, it would be something for us to do, along with other flavours.